optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -984,5 +1007,54 @@ def _create_objective(
             self.lgbm_kwargs,
             self.best_score,
             step_name=step_name,
+            model_dir=self._model_dir,
             pbar=pbar,
         )
+
+    def get_best_booster(self) -> "lgb.CVBooster":
+        """Return the best cvbooster.
+
+        If the best booster cannot be found, :class:`ValueError` will be raised.
+        To prevent the errors, please save boosters by specifying
+        both of the ``model_dir`` and the ``return_cvbooster`` arguments of
+        :meth: `~optuna.integration.lightgbm.LightGBMTunerCV.__init__`
+        when you resume tuning or you run tuning in parallel.
+        """
+        if self.lgbm_kwargs.get("return_cvbooster") is not True:
+            raise ValueError(
+                "LightGBMTunerCV requires `return_cvbooster=True` for method `get_best_booster()`."
+            )
+        if self._best_booster_with_trial_number is not None:
```

Comment:
I found that `self._best_booster_with_trial_number` was always `None` since it was not set at `self._tune_params`. Now that both `LightGBMTuner` and `LightGBMTunerCV` uses `self._best_booster_with_trial_number`, how about adding the following line to the end of `_LightGBMBaseTuner._tune_params`?

```python
        if objective.best_booster_with_trial_number is not None:
            self._best_booster_with_trial_number = objective.best_booster_with_trial_number
```

Reply:
Thank you for the clarification and suggestion. I was misunderstanding how `_best_booster_with_trial_number` works, especially when `model_dir==None`.
Have added the line as you suggested, and updated test code as well.